src/shogun/multiclass/tree/CARTree.cpp

```diff
@@ -111,7 +111,29 @@ std::shared_ptr<MulticlassLabels> CARTree::apply_multiclass(std::shared_ptr<Feat
 	auto current=get_root()->as<bnode_t>();
 
 	require(current, "Tree machine not yet trained.");
-	return apply_from_current_node(data->as<DenseFeatures<float64_t>>(), current)->as<MulticlassLabels>();
+	if (auto subfeat_data =
+	        std::dynamic_pointer_cast<DenseSubSamplesFeatures<float64_t>>(data))
+	{
+		size_t num_feat = subfeat_data->get_dim_feature_space();
+		auto num_vec = subfeat_data->get_num_vectors();
+		SGMatrix<float64_t> feature_matrix(num_feat, num_vec);
+		for (size_t i = 0; i < num_vec; i++)
+		{
+			SGVector<float64_t> v =
+			    subfeat_data->get_computed_dot_feature_vector(i);
+			sg_memcpy(
+			    feature_matrix.data() + i * num_feat, v.data(),
```

Comment:
I think there is a method that returns the pointer to the i-th column vector, might make it easier to read?
I also think there is an iterator for column vectors, which might make it even more clean :)